metadata-ingestion/src/datahub/ingestion/source/mode.py

```diff
@@ -900,18 +919,24 @@ def normalize_mode_query(self, query: str) -> str:
 
         return rendered_query
 
-    def construct_query_from_api_data(
+    def construct_query_or_dataset(
         self,
         report_token: str,
         query_data: dict,
         space_token: str,
         report_info: dict,
+        is_mode_dataset: bool = False,
```

Comment:
always explicitly set this when calling using a kwarg

```suggestion
        is_mode_dataset: bool,
```